org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -290,6 +308,9 @@ public IStatus run(IProgressMonitor monitor) {
 					status = ServiceStatus.ServiceReady;
 					pm.projectsImported(monitor);
 
+					boolean firstTimeInitialization = ProjectUtils.getAllProjects().length == 0;
+					telemetryManager.onServiceReady(pm, System.currentTimeMillis(), firstTimeInitialization);
```

Comment:
At this moment, the project import has finished, and `ProjectUtils.getAllProjects().length` should never be zero?

Reply:
Good catch. The setting needs to be recorded early on.

Reply:
Moved to `telemetryManager.onLanguageServerStart(..)`. Initially I was concerned it might be too early, but the Java model seems to respond correctly.